plugins/hikvision/src/main.ts

```diff
@@ -26,13 +26,20 @@ class HikVisionCamera extends RtspSmartCamera implements Camera {
                 events.on('close', () => ret.emit('error', new Error('close')));
                 events.on('error', e => ret.emit('error', e));
                 events.on('event', (event: HikVisionCameraEvent, cameraNumber: string) => {
-                    // if (this.getRtspChannel() && cameraNumber !== this.getCameraNumber()) {
-                    //     return;
-                    // }
                     if (event === HikVisionCameraEvent.MotionDetected
                         || event === HikVisionCameraEvent.LineDetection
                         || event === HikVisionCameraEvent.FieldDetection) {
-                        this.motionDetected = true;
+                        if (this.getRtspChannel() && cameraNumber !== this.getCameraNumber()) {
+                            // this.console.error(`### Skipping motion event ${cameraNumber} != ${this.getCameraNumber()}`);
+                            return;
+                        }
+        
+                        // this.console.error('### Detected motion, camera: ', cameraNumber);
+                        const prevMotion = this.motionDetected;
+                        if (!prevMotion) { 
+                            this.motionDetected = true;
```

Comment:
Is there a reason for this? It shouldn't be necessary. Scrypted will automatically denoise no-op state changes. But it does log that the state was updated at a specific time.

Reply:
I did it just so a log event would be emitted only when an actual change occurred.  (I was getting overwhelmed with logs…)

Reply:
> I did it just so a log event would be emitted only when an actual change occurred. (I was getting overwhelmed with logs…)

I see, I'll remove it then. I prefer to have those events, as sent from the device, even if they're duplicating the state. I can denoise it in the event log viewer.